src/pages/Experiment.less

```diff
@@ -38,20 +27,39 @@
         color: @gray;
     }
 
-    &__progress {
-        margin-bottom: 20px;
-    }
-
     &__content {
-        flex-grow: 1;
-        display: flex;
-        flex-direction: column;
+        height: 100%;
     }
 
     &__diff {
-        flex-grow: 1;
-        display: flex;
-        margin-bottom: 20px;
+        height: 100%;
+
+        // this is the only way I know to create 100% height page
+        // with header & footer & scrollable content in the middle (without flexbox)
+        //
+        // The idea: wrapper for scrollable div has padding-bottom
+        // of the height of footer + header, so the scrollable div has correct max-height
+        // and than margin-bottom: -(padding-bottom) to lift up footer
+        // the page has overflow hidden to hide empty space at the bottom
+        // for better understanding look at codepen: https://codepen.io/smacker/pen/mXyPQE
+        //
+        // Calculating height
+        // NB!!!! layout will get broken if any of heights change
+        //
+        // must be sum of everything on top + footer
+        // page-header: 50px
+        // experiment-header: 23px
+        // margins: 20px * 2 + 10px
+        // footer: 46+10px
+        // borders: 2px
+        // 50 + 23 + 40 + 56 + 10 + 2 = 181px
+        padding-bottom: 181px;
```

Comment:
I understand the problem, but I think it should be faced with variables.
Header and footer sizes could be defined in vars, so that vars could be used here.
That way, changing something in another place, would not break this component.
Making the header fixed will mean another thing: it could break if its content is changed, but it should be a local problem, instead of an external problem.

Reply:
Thanks for looking in it. As I explained before variables don’t really save us from breaking layout. But moving some numbers to variables just for better readability might be a good idea.
Could you please provide a small snippet for your idea, so I will be sure I got what you mean right?

Reply:
yes, things will eventually break ;) I only tried to break things in a limited scope. Let's see if this can do it:

(code below this line, is kind of pseudocode :wink: )

header.less
```less
$headerInnerHeight: 33px;
$headerMargingBottom: 10px;
$headerHeight: $headerInnerHeight + $headerMargingBottom;
.header { height: $headerInnerHeight; margin-bottom: $headerMargingBottom;}
```

footer.less
```less
$footerInnerHeight: 22px;
$footerMargingBottom: 5px;
$footerHeight: $footerInnerHeight + $footerMargingBottom;
.header { height: $footerInnerHeight; margin-bottom: $footerMargingBottom;}
```

code-panel.less
```less
@import 'header.less'
@import 'footer.less'
.code-panel {
    height: 100%;
    padding-top: $headerHeight;
    padding-bottom: $footerHeight
}
```

That way the code-panel does not depend on hardcoded values, but in values defined in places where that values refer to.

Reply:
yeah. correct. Only 1 thing `height: $headerInnerHeight ` is confusing in header/footer. Because it's not what it does. I will proceed without this rule, but with variable+comment.

Reply:
I'm not sure if I understood you when you said
> _$headerInnerHeight is confusing in header/footer. Because it's not what it does._
Let's see your implementation; since mine was only pseudocode, it didn't considered the whole component definition :dancer: